pydantic/_internal/_config.py

```diff
@@ -83,6 +83,7 @@ class ConfigWrapper:
     regex_engine: Literal['rust-regex', 'python-re']
     validation_error_cause: bool
     use_attribute_docstrings: bool
+    json_schema_literal_type: Literal['enum', 'oneof-const']
```

Comment:
I'd been following [a great suggestion](https://github.com/pydantic/pydantic/issues/8888#issuecomment-1966674737) to look to the newly added opt-in attribute docstring support for inspiration here.

But I'm realizing that I followed that paradigm a bit too closely by putting the config option here, I think the best place for it is in a `BaseModel.model_json_schema()` parameter. Will refactor.

Reply:
@rmehyde,

Ah yeah, that makes sense re moving to `model_json_schema`. That being said, you should still be able to keep most / all of the docs / tests that you've written, which is nice. Ping me when you've refactored, and I'd be more than happy to review! Looks like great work so far :).

Reply:
Thanks @sydney-runkle, I've made that update!

There's one check failing: a segfault when running the Ubuntu 3.10 tests. I don't believe this is directly related to my changes, and I don't get that behavior running the command in a 3.10 environment on my own Ubuntu machine. But I did re-run the CI/CD and the behavior is consistent, let me know what the right next step there is.

Reply:
All good, that test is annoyingly flaky. I just reran it again, hopefully it passes this time 🍀

Reply:
At a first glance, looks great. I'll get back to you with a final confirmation tomorrow regarding if this is the best place to have this parameter. I think it is, but I suppose I could see an argument for just creating a custom instance of `GenerateJsonSchema` with this logic, and passing that to the `model_json_schema()` function.

Could you re-add that comprehensive docs section that you added? Maybe to the JSON schema docs (instead of where you originally had it, in the config docs)